diff --git a/talk/build/common.gypi b/talk/build/common.gypi
index 50acb39..ffef235 100644
--- a/talk/build/common.gypi
+++ b/talk/build/common.gypi
@@ -40,7 +40,7 @@
       ['OS=="android" or OS=="linux"', {
         # TODO(henrike): make sure waterfall bots have $JAVA_HOME configured
         # properly and remove the default value below. See issue 2113.
-        'java_home%': '<!(python -c "import os; dir=os.getenv(\'JAVA_HOME\', \'/usr/lib/jvm/java-6-sun\'); assert os.path.exists(os.path.join(dir, \'include/jni.h\')), \'Point \\$JAVA_HOME or the java_home gyp variable to a directory containing include/jni.h!\'; print dir")',
+        'java_home%': '<!(python -c "import os; java_path = [os.path.join(\'/usr/lib/jvm\', p) for p in [\'java-7-oracle\', \'java-7-openjdk-amd64\', \'java-7-openjdk\', \'java-6-openjdk-amd64\', \'java-6-openjdk\'] if os.path.exists(os.path.join(\'/usr/lib/jvm\', p))][0]; dir=os.getenv(\'JAVA_HOME\', java_path); assert os.path.exists(os.path.join(dir, \'include/jni.h\')), \'Point \\$JAVA_HOME or the java_home gyp variable to a directory containing include/jni.h!\'; print dir")',
       }],
     ],
   },
